### PR TITLE
refactor few main points

### PR DESCRIPTION
**Pull request by Hyaxia**
- authentication token now has 3 additional fields - companyId, institutionId, programId
- error handling is now more generic
- volunteer mapping now should handle cases when program manager or coordinator try to delete mapping
- updated tests to access the api instead of the use case function directly

### CODE REVIEW

File: src/app/user.ts [hunk 1]
```diff
@@ -22,7 +25,14 @@ export const login = async (userId: string): Promise<loginResponse> => {
   if (!userId || !user) {
     return { userDetails: null, isFound: false, userToken: '', userType: null };
   }
-  const tokenData: Partial<userDecoded> = { userType: user.userType, userId };
+  const tokenData: Partial<userDecoded> = {
+    userType: user.userType,
+    userId,
+    // institutionId, programId and companyId might be null when returned from the database
+    institutionId: user.institutionId || undefined,
```
Comment (Itay9263) [inline src/app/user.ts:32]: No need for undefined check

File: src/models/user.ts [hunk 2]
```diff
@@ -30,13 +30,13 @@ export class User {
   userType: UserType;
 
   @Column({ nullable: true })
-  institutionId!: number;
+  institutionId?: number | null;
```
Comment (Itay9263) [inline src/models/user.ts:33]: Isn't that the whole point of using the nullable in the decorator?
I'm not really sure though, you might need this for type safety

File: tests/setup.ts [hunk 3]
```diff
@@ -14,3 +18,15 @@ before(async () => {
 after(async () => {
   appDataSource.destroy();
 });
+
+// create test jwt using User object
+export const createTestJwt = (user: User): string => {
+  const payload = {
+    userId: user.id,
+    userType: user.userType,
+    institutionId: user.institutionId || undefined,
```
Comment (Itay9263) [inline tests/setup.ts:27]: Same here

File: src/models/user.ts (hunk 2 above)
Comment (Hyaxia) [reply src/models/user.ts:33]: yeah that's why I've added it

File: src/app/user.ts [hunk 4]
```diff
@@ -13,6 +13,9 @@ type loginResponse = {
 export type userDecoded = {
   userId: string;
   userType: UserType;
+  institutionId: number | undefined;
```
[the remark this answers is not in the document]
Comment (Hyaxia) [reply src/app/user.ts:16]: right, will change it

**Pull request merged**